enhancements/cloud-integration/infrastructure-external-platform-type.md

```diff
@@ -0,0 +1,560 @@
+
+In the context of this EP, no particular changes will be needed in the operator itself,
+changes made in [library-go](#library-go) with further dependency update should be suffice.
+
+#### Machine Config Operator
+
+Currently, MCO sets kubelet flags based on a [PlatformStatus and PlatformType](https://github.com/openshift/machine-config-operator/blob/7babc03f84088f49072c6f5b0d1769850d09a694/pkg/controller/template/render.go#L343).
+This flag is [crucial](https://kubernetes.io/docs/tasks/administer-cluster/running-cloud-controller/#running-cloud-controller-manager) for Cloud Controller Manager engagement within the cluster.
+
+Initially, the new "External" platform should be treated similarly to PlatformType "None" by the MCO, and do not set up any cloud-specific flags for the kubelet.
+Then, down the road (during [phase 3](#implementation-phases)), it would be expected for the MCO to use the "External" platform type and its spec as a signal about the underlying platform and cloud controller manager presence and operate accordingly.
+For an explicit signal about the necessity to set `--cloud-provider=external` flag to the kubelet, we will use the `CloudControllerManager` field of the `ExternalPlatformSpec`, which is  
+described in the [API Extensions](#api-extensions) section down below.
+
+#### Cluster Cloud Controller Manager Operator
+
+Responsible for deploying platform-specific Cloud Controller Manager as well as for handling a number of OpenShift's
+specific peculiarities (such as populating proxy settings for CCMs, sync credentials, and so on).
+
+The code from the [library-go](#library-go) using for decision-making about operator engagement.
+In case library-go's [IsCloudProviderExternal](https://github.com/openshift/library-go/blob/e1213f6ec5d10aa4aa8a4cac2780b1dc674c0396/pkg/operator/configobserver/cloudprovider/observe_cloudprovider.go#L154) function
+indicates that cloud provider is external and the operator encounters a platform which it is not aware of it will go into 'Degraded' state.
+
+During the first phases of the "External" platform type enablement, this operator should be just disabled.
+This might be done with changes within the [library-go](#library-go) and further dependency updates or, better,
+by adding a respective check within the operator itself.
+
+In the future, when the delivery mechanism for CCMs will be defined,
+the operator might be engaged for deploying a user-provided cloud controller manager,
+however this is a subject for the upcoming design work.
+
+#### Machine Api Operator
+
+Machine Api Operator is responsible for deploying and maintaining the set of machine-api related controllers, such as
+
+- machineset controller
+- nodelink controller
+- machine health check controller
+- machine controller
+
+From the list above, only the "machine controller" is cloud-provider dependent, however, for now, Machine Api Operator
+won't deploy anything if it encounters "None" or an unrecognized platform type. 
```

Comment:
In the *Background* section today you have:

```markdown
Additionally, there are difficulties that are present today due to having a predefined list of platforms.
A few examples:
- No defined mechanism to set `--cloud-provider=external` arg to kubelet/KCM/apiserver without merges and further revendoring of "openshift/api",
  at the moment decision-making here is tied to the PlatformType.
```

which is basically the issue with `None`, as I understand it, but it might be worth a clearly-titled subsection under *Alternatives* spelling out this `None` limitation, that you could link from these other places where you assert `None` is not sufficient.

Reply:
Extended _Alternatives_ section: https://github.com/openshift/enhancements/pull/1234/commits/92fcbed245f485af0c8baf9ed3bd07b7d874b987#diff-08391a193550547988305aefa40e474c593e97d3b2375245c31015f39f5812b3R556